docs/source/task-launch.rst

```diff
@@ -84,67 +117,122 @@ We can submit tasks from other tasks.  This allows us to make decisions while
 on worker nodes.
 
 To submit new tasks from a worker that worker must first create a new client
-object that connects to the scheduler.  There is a convenience function to do
-this for you so that you don't have to pass around connection information.
-However you must use this function ``worker_client`` as a context manager to
-ensure proper cleanup on the worker.
+object that connects to the scheduler. There are three options for this:
+
+1. ``dask.compute`` and ``dask.delayed``
+2. ``get_client`` with ``secede`` and ``rejoin``
+3. ``worker_client``
+
+We demonstrate the use of these functions below. But before we do, here's some
+design considerations for each function:
+
+* ``dask.compute``
+    * After jobs collected from cluster, computation continues on the
+      scheduling slot that submitted the jobs.
+    * If using Dask objects, will perform some graph merging to perform less
+      computation and communication. See "`The compute function`_" for more
+      detail.
+* ``get_client``
+    * Will continue to occupy a scheduling slot, unless ``secede`` and
+      ``rejoin`` are called (otherwise may deadlock the cluster!).
+    * The worker running the function will possibly transition to a different
+      scheduling slot (the worker that submitted the jobs is labeled as "maybe
+      long running").
+* ``worker_client``
+    * Intended to submit long running jobs (the worker is labeled as "long
+      running")
+    * Implemented as a context manager, which provides some ease of use.
+    * Will ``secede`` from the current worker, and likely transition to another
+      worker.
+    * Establishing a connection to the scheduler takes roughly 10–20ms. It's
+      wise for the submitted jobs to be several times longer than this.
```

Comment:
Many of the details here assume that the reader already knows a lot about how these things work.  I might leave most of these notes for later.  Instead I would start with "here are three options and how to use them" then maybe "here is how you might choose between those options" finally finishing with "by the way, here are some technical details that you should beware of if you've made it this far in the document.